[Bug]: NullPointerException when editing unit test without selecting any unit test in Hop GUI

### Apache Hop version?

2.16.0

### Java version?

openjdk 17

### Operating system

Windows

### What happened?

When working in Hop GUI, if no unit test is selected and the user clicks “Edit selected unit test”, the application throws a NullPointerException instead of handling the action gracefully.

```txt
2026/01/15 09:02:32 [Error]    GUI - Caused by: java.lang.NullPointerException: Cannot invoke "org.apache.hop.testing.PipelineUnitTest.getName()" because "unitTest" is null
2026/01/15 09:02:32 [Error]    GUI - 	at org.apache.hop.testing.gui.TestingGuiPlugin.editUnitTest(TestingGuiPlugin.java:903)
```

### Issue Priority

Priority: 3

### Issue Component

Component: Hop Gui

--- a/plugins/misc/testing/src/main/java/org/apache/hop/testing/gui/TestingGuiPlugin.java
+++ b/plugins/misc/testing/src/main/java/org/apache/hop/testing/gui/TestingGuiPlugin.java
@@ -900,7 +900,9 @@ public class TestingGuiPlugin {
             hopGui.getMetadataProvider(),
             PipelineUnitTest.class,
             hopGui.getShell());
-    if (manager.editMetadata(unitTest.getName())) {
+    if (unitTest != null
+        && !Utils.isEmpty(unitTest.getName())
+        && manager.editMetadata(unitTest.getName())) {
       // Activate the test
       refreshUnitTestsList();
       selectUnitTest(pipelineMeta, unitTest);
@@ -1292,6 +1294,9 @@ public class TestingGuiPlugin {
 
     try {
       PipelineUnitTest unitTest = getUnitTestFromContext(context);
+      if (unitTest == null) {
+        return;
+      }
       PipelineUnitTestTweak unitTestTweak = unitTest.findTweak(transformMeta.getName());
       if (unitTestTweak != null) {
         unitTest.getTweaks().remove(unitTestTweak);
@@ -1401,7 +1406,7 @@ public class TestingGuiPlugin {
   public void switchUnitTest(PipelineUnitTest targetTest, PipelineMeta pipelineMeta) {
     try {
       TestingGuiPlugin.getInstance().detachUnitTest();
-      TestingGuiPlugin.getInstance().selectUnitTest(pipelineMeta, targetTest);
+      TestingGuiPlugin.selectUnitTest(pipelineMeta, targetTest);
     } catch (Exception exception) {
       new ErrorDialog(
           HopGui.getInstance().getShell(),

